packages/effect/src/Cron.ts

```diff
@@ -285,66 +338,99 @@ export const match = (cron: Cron, date: DateTime.DateTime.Input): boolean => {
  * @since 2.0.0
  */
 export const next = (cron: Cron, now?: DateTime.DateTime.Input): Date => {
-  const { days, hours, minutes, months, weekdays } = cron
+  const zoned = dateTime.unsafeMakeZoned(now ?? new Date(), {
+    timeZone: Option.getOrUndefined(cron.tz)
+  })
 
-  const restrictMinutes = minutes.size !== 0
-  const restrictHours = hours.size !== 0
-  const restrictDays = days.size !== 0
-  const restrictMonths = months.size !== 0
-  const restrictWeekdays = weekdays.size !== 0
+  const adjustDst = dateTime.isUtc(zoned) ? constVoid : (current: Date) => {
```

Comment:
.isUtc will always be false here

Reply:
Gotcha. The intent here was to check if the Cron instance is set to UTC. I adapted it to cover that now.